autotests/testkeyverification.cpp

```diff
@@ -15,23 +17,37 @@ private Q_SLOTS:
     void testVerification()
     {
         CREATE_CONNECTION(a, "alice1", "secret", "AliceDesktop")
-        CREATE_CONNECTION(b, "alice1", "secret", "AlicePhone")
+        CREATE_CONNECTION(b, "alice2", "secret", "AlicePhone")
+
+        a->requestDirectChat(b->userId());
+            connectSingleShot(b.get(), &Connection::invitedRoom, this, [=](Quotient::Room* room, Quotient::Room*) {
+            b->joinRoom(room->id());
+            a->room(room->id())->activateEncryption();
+        });
 
         QPointer<KeyVerificationSession> aSession{};
         connect(a.get(), &Connection::newKeyVerificationSession, this, [&](KeyVerificationSession* session) {
             aSession = session;
+            QVERIFY(aSession);
+            connect(session, &KeyVerificationSession::stateChanged, this, [session](){
+                QVERIFY(session->state() != KeyVerificationSession::CANCELED);
+            });
             QVERIFY(session->remoteDeviceId() == b->deviceId());
             QVERIFY(session->state() == KeyVerificationSession::WAITINGFORREADY);
             connectSingleShot(session, &KeyVerificationSession::stateChanged, this, [=](){
                 QVERIFY(session->state() == KeyVerificationSession::ACCEPTED || session->state() == KeyVerificationSession::READY);
                 connectSingleShot(session, &KeyVerificationSession::stateChanged, this, [=](){
                     QVERIFY(session->state() == KeyVerificationSession::WAITINGFORVERIFICATION);
-                    session->sendMac();
                 });
             });
         });
-        a->startKeyVerificationSession(b->deviceId());
-        connect(b.get(), &Connection::newKeyVerificationSession, this, [=](KeyVerificationSession* session) {
+        QTimer::singleShot(3000, this, [a, b](){
```

Comment:
Can I have a comment in the code here on why the delay is needed?

Also:
```suggestion
        QTimer::singleShot(3000, this, [a, b] {
```